pallets/duster/src/lib.rs

```diff
@@ -0,0 +1,263 @@
+		type MultiCurrency: MultiCurrencyExtended<
+			Self::AccountId,
+			CurrencyId = Self::CurrencyId,
+			Balance = Self::Balance,
+			Amount = Self::Amount,
+		>;
+
+		/// The minimum amount required to keep an account.
+		type MinCurrencyDeposits: GetByKey<Self::CurrencyId, Self::Balance>;
+
+		/// Account to send dust to.
+		#[pallet::constant]
+		type DustAccount: Get<Self::AccountId>;
+
+		/// Account to take reward from.
+		#[pallet::constant]
+		type RewardAccount: Get<Self::AccountId>;
+
+		/// Reward amount
+		#[pallet::constant]
+		type Reward: Get<Self::Balance>;
+
+		/// Native Asset Id
+		#[pallet::constant]
+		type NativeCurrencyId: Get<Self::CurrencyId>;
+
+		/// Weight information for extrinsics in this module.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub account_blacklist: Vec<T::AccountId>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			GenesisConfig {
+				account_blacklist: vec![],
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			self.account_blacklist.iter().for_each(|account_id| {
+				AccountBlacklist::<T>::insert(account_id, ());
+			})
+		}
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Account is excluded from dusting.
+		AccountBlacklisted,
+
+		/// Account is not present in the non-dustable list.
+		AccountNotBlacklisted,
+
+		/// The balance is zero.
+		ZeroBalance,
+
+		/// The balance is sufficient to keep account open.
+		BalanceSufficient,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Account dusted.
+		Dusted(T::AccountId, T::Balance),
+
+		/// Account added to non-dustable list.
+		Added(T::AccountId),
+
+		/// Account removed from non-dustable list.
+		Removed(T::AccountId),
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight((<T as Config>::WeightInfo::dust_account(), DispatchClass::Normal, Pays::Yes))]
+		pub fn dust_account(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			currency_id: T::CurrencyId,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			ensure!(Self::blacklisted(&account).is_none(), Error::<T>::AccountBlacklisted);
+
+			let (dustable, dust) = Self::is_dustable(&account, currency_id);
+
+			ensure!(dust != T::Balance::from(0u32), Error::<T>::ZeroBalance);
+
+			ensure!(dustable, Error::<T>::BalanceSufficient);
+
+			Self::transfer_dust(&account, &T::DustAccount::get(), currency_id, dust)?;
+
+			if currency_id == T::NativeCurrencyId::get() {
+				// Not sure if we should fail here?
+				let _ = frame_system::Pallet::<T>::dec_providers(&account);
+			}
+
+			Self::deposit_event(Event::Dusted(account, dust));
+
+			// Ignore the result, it fails - no problem.
+			let _ = Self::reward_duster(&who, currency_id, dust);
+
+			Ok(().into())
```

Comment:
You might want to refund the fees if the dusting was successful. That way you can maybe get away with a smaller reward.